libs/vkd3d/state.c

```diff
@@ -4366,12 +4382,30 @@ static HRESULT d3d12_pipeline_state_init_static_pipeline(struct d3d12_pipeline_s
          * be exceedingly rare. */
         can_compile_pipeline_early =
                 (desc->primitive_topology_type != D3D12_PRIMITIVE_TOPOLOGY_TYPE_PATCH || graphics->patch_vertex_count != 0 ||
-                 state->device->device_info.extended_dynamic_state2_features.extendedDynamicState2PatchControlPoints) &&
-                 desc->primitive_topology_type != D3D12_PRIMITIVE_TOPOLOGY_TYPE_UNDEFINED;
+                 state->device->device_info.extended_dynamic_state2_features.extendedDynamicState2PatchControlPoints);
```

Comment:
I think there's a flaw here if patch_vertex_count == 0 and dynamic patch controls are not supported. We're the in a situation where we cannot compile pre-raster state. I think we can just promote 0 to 1 here. 0 will only happen if there are no vertex inputs, and those shaders tend to just report 1 anyways. That would neatly bypass all this jank.